includes/drush.inc

```diff
@@ -308,6 +308,7 @@ function drush_get_global_options($brief = FALSE) {
     $options['confirm-rollback']    = array('description' => 'Wait for confirmation before doing a rollback when something goes wrong.');
     $options['complete-debug']      = array('hidden' => TRUE, 'description' => "Turn on debug mode forf completion code");
     $options['php-options']         = array('description' => "Options to pass to `php` when running drush.  Only effective when using the drush.launcher script.", 'never-propagate' => TRUE, 'example-value' => '-d error_reporting="E_ALL"');
+    $options['halt-on-error']       = array('propagate' => TRUE, 'description' => "Controls error handling of recoverable errors (E_RECOVERABLE_ERROR). --halt-on-error=1: Execution is halted. --halt-on-error=0: Drush execution continues");
```

Comment:
`propagate-cli-value` would be better here. With `propagate`, you are more likely to cause Drush to pass this value to a remote Drush, which might be older and not recognize the setting.  With `propagate-cli-value`, the option is only included if it appears on the command line.  Presumably, the value should be read again by the subprocess, if it came from a configuration file in the first place.